src/const_eval.rs

```diff
@@ -0,0 +1,63 @@
+use rustc::hir::def_id::DefId;
+use rustc::traits::Reveal;
+use rustc::ty::subst::Substs;
+use rustc::ty::{self, TyCtxt};
+
+use error::{EvalError, EvalResult};
+use lvalue::{Global, GlobalId, Lvalue};
+use rustc_const_math::ConstInt;
+use eval_context::{EvalContext, StackPopCleanup};
+
+pub fn eval_body<'a, 'tcx>(
+    tcx: TyCtxt<'a, 'tcx, 'tcx>,
+    (def_id, substs): (DefId, &'tcx Substs<'tcx>),
+) -> EvalResult<'tcx, ConstInt> {
```

Comment:
Hmm, that seems plausible. Could you change the name of this function to `eval_body_as_integer`?